src/sentry/api/serializers/models/projectcodeowners.py

```diff
@@ -1,9 +1,24 @@
 from sentry.api.serializers import Serializer, register
 from sentry.models import ProjectCodeOwners
+from sentry.models.integration import Integration
 
 
 @register(ProjectCodeOwners)
 class ProjectCodeOwnersSerializer(Serializer):
+    def get_attrs(self, item_list, user, **kwargs):
+        attrs = {}
+        for item in item_list:
+            org_integration = item.repository_project_path_config.organization_integration
+            if org_integration:
+                try:
+                    integration = Integration.objects.get(id=org_integration.integration_id)
+                    attrs[item] = {"provider": integration.provider}
+                except Integration.DoesNotExist:
+                    attrs[item] = {"provider": "unknown"}
+            else:
+                attrs[item] = {"provider": "unknown"}
```

Comment:
This actually still makes N queries, because you're still traversing the path here individually.

You can actually use `sentry.utils.db.attach_foreignkey` to help out here. Typically, we'd use `.select_related` to cache these rows, but inside the serializers we don't have a queryset, we just have a list of model instances.

You can use this like `attach_foreignkey(item_list, ProjectCodeOwners. repository_project_path_config, related=("organization_integration", ))`

What this does is cache the `.` relations on the models. So all you need to do is call this and the caching should be done. Then you can just do 
```
        if obj.organization_integration:	
            data["provider"] = obj.organization_integration.integration.provider
```
in `serialize` like you were before.

Reply:
@wedamija after calling `attach_foreignkey`, I think we still need to access the provider like `obj.repository_project_path_config.organization_integration.integration.provider`.